app/uk/gov/hmrc/agentsexternalstubs/controllers/PersonalDetailsValidationController.scala

```diff
@@ -16,14 +17,25 @@ class PersonalDetailsValidationController @Inject()(cc: ControllerComponents)(im
 
   import uk.gov.hmrc.agentsexternalstubs.models.PersonalDetailsValidationFormat._
 
+  private val validationIds = mutable.Map[String, Boolean]()
+
   def get(id: String): Action[AnyContent] = Action { implicit request =>
     Ok(toJson(personalDetailsValidationRepository.get(id)))
   }
 
+  // associate a result (success/failure) with a validation id
+  def pdvResult(id: String, success: Boolean): Action[AnyContent] = Action {
+    validationIds(id) = success
+    Ok
+  }
+
   object personalDetailsValidationRepository {
     def get(id: String): PersonalDetailsValidation =
-      PersonalDetailsValidation
-        .successful(id, PersonalDetailsWithNino("Fred", "Bloggs", LocalDate.now(), Generator.ninoNoSpaces(id)))
+      if (validationIds.getOrElse(id, false))
+        PersonalDetailsValidation
+          .successful(id, PersonalDetailsWithNino("Fred", "Bloggs", LocalDate.now(), Generator.ninoNoSpaces(id)))
```

Comment:
why don't we get this from the user record collection?  does this Generator genrates the same nino every time for the same `id` ?